analysis/bin/vtsAnalyser.cc

```diff
@@ -239,24 +360,25 @@ void vtsAnalyser::HadronAnalysis() {
         if ( had_chi2[k] > 3 ) continue;
         if ( had_dca[k] > 1 ) continue;
       }
+/*
       else if (abs(had_pdgId[k]) == 3122) {
         if ( (had_lxy[k]/had_lxyErr[k]) < 3 ) continue;
         if ( had_angleXY[k] < 0.98 ) continue;
         if ( had_chi2[k] > 3 ) continue;
         if ( had_dca[k] > 1 ) continue;
       }
+*/
       else continue;
       TLorentzVector had_tlv;
       had_tlv.SetPtEtaPhiM(had_pt[k], had_eta[k], had_phi[k], had_mass[k]);
       if (jet_tlv.DeltaR(had_tlv) < 0.3 && (had_pt[k]/Jet_pt[j]) > 0.15) {
-        Stat.idx = k;
-        Stat.pdgId = had_pdgId[k];
-        Stat.x = had_pt[k]/Jet_pt[j];
-        if (m_isMC) Stat.label = qjMapForMC_[j];
-        Stat.isHadJetMatched = true;
-        Stat.dr = jet_tlv.DeltaR(had_tlv);
-        Stat.jetIdx = j;
-        Had.push_back(Stat);
+        if (m_isMC) {
+          HadStat Stat(k, had_pdgId[k], qjMapForMC_[j], j, had_pt[k]/Jet_pt[j], jet_tlv.DeltaR(had_tlv), true);
```

Comment:
If you want to create the structure here, do it in the vector push_back so you can use the move semantics to only create it once (it will be copied in the push_back on the next line, I think)